photutils/psf/utils.py

```diff
@@ -342,3 +381,158 @@ def _interpolate_missing_data(data, mask, method='cubic'):
     data_interp[mask] = interpol(xy_missing)
 
     return data_interp
+
+
+def grid_from_epsfs(epsfs, fiducials=None, meta=None):
+
+    """
+    Create a GriddedPSFModel from a list of EPSFModels.
+
+    Given a list of EPSFModels, this function will return a GriddedPSFModel.
+    The fiducial points for each input EPSFModel can either be set on each
+    individual model by setting the 'x_0' and 'y_0' attributes, or provided as
+    a list of tuples (``fiducials``). If a ``fiducals`` list is provided, it
+    must match the length of input EPSFs. In either case, the fiducial points
+    must be on a grid.
+
+    Optionally, a ``meta`` dictionary may be provided for the
+    output GriddedPSFModel. If this dictionary contains the keys 'grid_xypos',
+    'oversampling', or 'fill_value', they will be overridden.
+
+    Note: If set on the input EPSFModel (x_0, y_0), then ``origin`` must be the
+    same for each input EPSF. Additionally data units and dimensions must be
+    for each input EPSF, and values for ``flux`` and ``oversampling``, and
+    ``fill_value`` must match as well.
+
+    Parameters
+    ----------
+    epsfs : list of `photutils.psf.models.EPSFModel`
+        A list of EPSFModels representing the individual PSFs.
+    fiducials : list, optional
+        A list of fiducial points (x_0, y_0) for each PSF.
+        If not provided, the x_0 and y_0 of each input EPSF will be considered
+        the fiducial point for that PSF. Default is None.
+    meta : dict, optional
+        Additional metadata for the GriddedPSFModel.
+        Default is None.
```

Comment:
Mention that certain keys, e.g., `grid_xypos`, (list them all) will be overridden.